sdk/aqueduct/artifacts/generic_artifact.py

```diff
@@ -7,34 +7,36 @@
 from aqueduct.artifacts.base_artifact import BaseArtifact
 from aqueduct.dag import DAG
 from aqueduct.enums import ArtifactType
-from aqueduct.error import AqueductError
+from aqueduct.error import AqueductError, InvalidArtifactTypeException
 
 
 class GenericArtifact(BaseArtifact):
     """This class represents a generic artifact within the flow's DAG.
-    Currently, a generic artifact can be any artifact other than table, numeric, bool, or parameter.
+    Currently, a generic artifact can be any artifact other than table, numeric, bool, or parameter
+    generated from eager execution, or an artifact of unknown type generated from lazy execution.
     """
 
     def __init__(
         self,
         dag: DAG,
         artifact_id: uuid.UUID,
-        type: ArtifactType,
+        artifact_type: ArtifactType = ArtifactType.UNTYPED,
         content: Optional[Any] = None,
         from_flow_run: bool = False,
     ):
         self._dag = dag
         self._artifact_id = artifact_id
         # This parameter indicates whether the artifact is fetched from flow-run or not.
         self._from_flow_run = from_flow_run
-        self._content = content
+        self.set_type(artifact_type)
+        self.set_content(content)
+
         if self._from_flow_run:
             # If the artifact is initialized from a flow run, then it should not contain any content.
-            assert self._content is None
-        else:
-            assert self._content is not None
+            assert self.content() is None
```

Comment:
@HarisChoudhary I'm assuming you are fixing this as part of your parameter work, as we discussed that for artifact generated from flow run, its content should already be available via calling the backend's `get_artifact_result` endpoint. So the correct logic would be if `from_flow_run` is true, then content should never be `None`.